syntax = "proto3";

package similarity;

service SimilarityService {
    rpc ComputeSimilarity (SimilarityRequest) returns (SimilarityResponse);
    rpc SparseEmbedding (SimilarityRequest) returns (SimilarityResponse);
    rpc ColBERT (SimilarityRequest) returns (SimilarityResponse);
}

message SimilarityRequest {
    repeated string sentences_1 = 1;
    repeated string sentences_2 = 2;
}

message SimilarityResponse {
    repeated float similarity = 1;
}